sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -322,16 +412,18 @@ def _generate_pythonop(component_func, target_image):
     component_artifact['implementation']['dockerContainer']['arguments'].append({'value': input})
   return _create_task_factory_from_component_dict(component_artifact)
 
-def build_python_component(component_func, staging_gcs_path, target_image, build_image=True, timeout=600, namespace='kubeflow'):
+def build_python_component(component_func, staging_gcs_path, target_image, build_image=True, timeout=600, namespace='kubeflow', dependency={}):
   """ build_component automatically builds a container image for the component_func
   based on the base_image and pushes to the target_image.
 
   Args:
     component_func (python function): The python function to build components upon
     staging_gcs_path (str): GCS blob that can store temporary build files
+    target_image (str): target image path
+    build_image (bool): whether to build the image or not. Default is True.
     timeout (int): the timeout for the image build(in secs), default is 600 seconds
     namespace (str): the namespace within which to run the kubernetes kaniko job, default is "kubeflow"
-    build_image (bool): whether to build the image or not. Default is True.
+    dependency (dict): a dictionary with key as the package name and value as the DependencyVersion, default is empty
```

Comment:
... `DependencyVersion` could just be a min/max version tuple:
```
{
  tensorflow: (0.10.0, 0.11.0),
  kubernetes: (0.6.0, ),
}
```

Reply:
I moved the name into the VersionedDependency as you suggested. I think having a class wrapping information and providing access interfaces are important for error handling, easy code read, and future code reuse.